plugin/src/main/java/org/opensearch/ml/action/forward/TransportForwardAction.java

```diff
@@ -123,20 +153,16 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<MLForw
                             modelState = MLModelState.DEPLOY_FAILED;
                             log.error("deploy model failed on all nodes, model id: {}", modelId);
                         }
+                        Map<String, Object> updateFields = new HashMap<>();
+                        updateFields.put(MLModel.MODEL_STATE_FIELD, modelState);
+                        updateFields.put(MLModel.LAST_DEPLOYED_TIME_FIELD, Instant.now().toEpochMilli());
+                        updateFields.put(MLModel.CURRENT_WORKER_NODE_COUNT_FIELD, currentWorkerNodeCount);
+                        if (calculatedRetryTimes != null) {
+                            log.debug("Calculated auto redeploy retry times is not null, setting it to 0");
```

Comment:
What does this mean? why we need to set to 0 if retry times is not null? Are we expecting retry times to be null? 
I don't expect customer will understand this message in their log. 

In addition, looks like from `processModelAutoRedeploy` we only return 0/null. So I don't understand the significance of null check and setting to 0 anyways.

Reply:
I'll change the method to return boolean value and a proper name to make it more clear.